### PR TITLE
Replaced dominant-baseline with dy to fix alignment issues on IE

### PR DESCRIPTION
**Pull request by jdhodges**
#734 #735 #738

### CODE REVIEW

File: src/assets/js/chart/legend.js [hunk 1]
```diff
@@ -33,7 +34,11 @@ export default function() {
 
             if (model.data) {
                 tooltipContainer.datum(model.data)
-                    .call(tooltip);
+                    .call(tooltip)
+                    .selectAll('text')
```
Comment (jleft) [inline src/assets/js/chart/legend.js]: Can this be done with `decorate()` on the tooltip, so the enter selection can be used?

File: src/assets/js/chart/legend.js [hunk 2]
```diff
@@ -33,7 +34,11 @@ export default function() {
...
+                    .each(function() {
+                        this.setAttribute('dy', textYOffset);
```
Comment (jleft) [inline src/assets/js/chart/legend.js]: Any reason not to use `.attr()`?

File: src/assets/js/chart/nav.js [hunk 3]
```diff
@@ -27,6 +28,11 @@ export default function() {
         .yTicks(0)
         .margin({
             bottom: bottomMargin      // Variable also in navigator.less - should be used once ported to flex
+        })
+        .decorate(function(selection) {
+            selection.enter().selectAll('.x-axis text').each(function() {
+                this.setAttribute('dy', xAxisTextYOffset);
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Any reason not to use `.attr()`?

File: src/assets/js/chart/primary.js [hunk 4]
```diff
@@ -205,6 +206,10 @@ export default function() {
                 });
               closePriceTick.select('text')
                 .attr('transform', 'translate(' + calloutHeight / 2 + ',1)');
+
+              s.selectAll('text').each(function() {
+                  this.setAttribute('dy', yAxisTextYOffset);
```
Comment (jleft) [inline src/assets/js/chart/primary.js]: Any reason not to use `.attr()`?

Could the enter selection be used, or does it need to be on the update selection?

File: src/assets/styles/axis.less [hunk 5]
```diff
@@ -41,11 +41,12 @@
     font-weight: @font-weight-base;
     line-height: @line-height-base;
     fill: @font-colour;
+    dominant-baseline: no-change;
```
Comment (jleft) [inline src/assets/styles/axis.less]: Why use `no-change` over `auto` or `middle`? (Same for all occurrences of dominant-baseline.)

File: src/assets/styles/axis.less [hunk 6]
```diff
@@ -41,11 +41,12 @@
...
   }
   .tick path {
     stroke: @gridline-colour;
   }
   .domain {
     stroke: @gridline-colour;
   }
-}
+}
```
Comment (jleft) [inline src/assets/styles/axis.less]: Re-add new line

File: src/assets/styles/legend.less [hunk 7]
```diff
@@ -14,6 +14,11 @@
     width: 500px;
     height: 20px;
     justify-content: center;
+    overflow: visible;
+
+    .cell{
```
Comment (jleft) [inline src/assets/styles/legend.less]: Space after `.cell`; before `{`

File: src/assets/js/chart/primary.js (hunk 4 above)
Comment (jdhodges) [inline src/assets/js/chart/primary.js]: This needs to be on update I think, can't get it to work properly on enter.

File: src/assets/js/chart/legend.js [hunk 8]
```diff
@@ -14,7 +15,14 @@ export default function() {
             ['L', function(d) { return priceFormat(d.low); }],
             ['C', function(d) { return priceFormat(d.close); }],
             ['V', function(d) { return volumeFormat(d.volume); }]
-        ]);
+        ])
+        .decorate(function(selection) {
+            selection.enter()
+                .selectAll('text')
+                .each(function() {
```
Comment (jleft) [inline src/assets/js/chart/legend.js]: Sorry - from my previous comment - I meant can the `each` be replaced with`attr`, e.g.:

```js
selection.enter()
    .selectAll('text')
    .attr('dy', textYOffset);
```

(For all occurrences of `each` in this PR.)

File: src/assets/js/chart/primary.js (hunk 4 above)
Comment (jleft) [inline src/assets/js/chart/primary.js]: :+1:

**Comment by jleft** [comment]
Axis label issues have been fixed by #792.

Legend fixes rebased in #795

**Pull request closed without merging**